scripts/linters/custom_eslint_checks/rules/no-testonly.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint to ensure that functions with 'testOnly'
+ * in their names are only called from 'spec' files.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: (
+        'Functions with "testOnly" in their names should be only ' +
+        'called from "spec" files'),
+      category: 'Best Practices',
+      recommended: true,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      notTestOnlyAllowed: 'Can not call a function containing \'testOnly\' ' +
+        'in a JS/TS file that is not a .spec file'
+    }
+  },
+  create: function(context) {
+    return {
+      CallExpression(node) {
+        if (typeof node.callee.name === 'string' &&
+          node.callee.name.toLowerCase().includes('testonly')) {
+          context.report({
+            node,
+            loc: node.loc,
+            messageId: 'notTestOnlyAllowed'
+          });
+        }
+      }
+    };
+  }
+};
```

Comment:
Where are you checking if the function is called from a non-test file? @DubeySandeep are we still using testOnly in a function name because I didn't find any example?

Reply:
@Hudda 
In the **.eslintrc** I have excluded the .spec files for this lint check. 
Why?
The idea of this lint is that functions with testOnly are not called in non-spec files. So in the lint rule I only check that the called functions do not contain TestOnly. And the scope of this rule will be just the no-spec files.

Reply:
To test that it works I do a temporary refactoring. 

For example here I change the name of a function to one that contains TestOnly.
![Screenshot from 2021-04-13 01-40-23](https://user-images.githubusercontent.com/46513107/114515573-1f10e280-9c02-11eb-84b3-87b88b34e12c.png)

In this case this function is called in 122 files, 121 .spec files (the yellow ones) and 1 non-spec file (the blue one). 
![Screenshot from 2021-04-13 01-45-20](https://user-images.githubusercontent.com/46513107/114515593-246e2d00-9c02-11eb-8c6d-ff24b00e7a25.png)

And finally the check only throws the error in the non-spec file, as it should be.
![Screenshot from 2021-04-13 01-44-46](https://user-images.githubusercontent.com/46513107/114515619-29cb7780-9c02-11eb-96c7-08ea22ef27cc.png)

Reply:
PTAL @Hudda